src/games/rcll/orders.clp

```diff
@@ -41,19 +41,22 @@
 
 (defrule order-recv-SetOrderDelivered
   (gamestate (state RUNNING) (phase PRODUCTION) (game-time ?gt))
-  ?pf <- (protobuf-msg (type "llsf_msgs.SetOrderDelivered") (ptr ?p) (rcvd-via STREAM)
-		       (rcvd-from ?from-host ?from-port) (client-id ?cid))
+  ?pf <- (protobuf-msg (type "llsf_msgs.SetOrderDelivered") (ptr ?p) (rcvd-via STREAM))
   =>
-  (bind ?id (pb-field-value ?p "order_id"))
-  (bind ?team (sym-cat (pb-field-value ?p "team_color")))
-  (if (not (any-factp ((?o order)) (= ?o:id ?id)))
+  (if (not (do-for-fact
```

Comment:
From the docs:

> The return value is the evaluation of the action. If no fact-set satisfied the query, then the return value is the symbol FALSE.

While I couldn't find what exactly the "evaluation of the action" means (it's not the return value), my understanding is that an action never evaluates to FALSE (possibly that's not true in an error case).

Reply:
This is correct usage of `do-for-fact`: https://stackoverflow.com/q/54367042/2785041

Reply:
Interesting, I'd still not use this (it is not obvious and kinda un-CLIPS-y for me, since it uses a procedural structure in a condition. But I'll not block the review about this.

Reply:
It avoids code duplication and therefore avoids difficult bugs if the condition changes but is changed in only one of the two places, so I think it's a good pattern to use.